src/main/java/com/puppycrawl/tools/checkstyle/JavaParser.java

```diff
@@ -238,26 +222,37 @@ private static DetailAST createCommentAstFromToken(Token token) {
      * @param token to create the AST
      * @return DetailAST with SINGLE_LINE_COMMENT type
      */
-    private static DetailAST createSlCommentNode(Token token) {
+    private static DetailAST createSlCommentNode(CommonToken token) {
         final DetailAstImpl slComment = new DetailAstImpl();
         slComment.setType(TokenTypes.SINGLE_LINE_COMMENT);
         slComment.setText("//");
 
-        // column counting begins from 0
-        slComment.setColumnNo(token.getColumn() - 1);
+        slComment.setColumnNo(token.getCharPositionInLine());
         slComment.setLineNo(token.getLine());
 
         final DetailAstImpl slCommentContent = new DetailAstImpl();
         slCommentContent.setType(TokenTypes.COMMENT_CONTENT);
 
-        // column counting begins from 0
         // plus length of '//'
-        slCommentContent.setColumnNo(token.getColumn() - 1 + 2);
+        slCommentContent.setColumnNo(token.getCharPositionInLine() + 2);
         slCommentContent.setLineNo(token.getLine());
         slCommentContent.setText(token.getText());
 
         slComment.addChild(slCommentContent);
         return slComment;
     }
 
+    /**
+     * Custom error listener to provide detailed exception message.
+     */
+    private static class CheckstyleErrorListener extends BaseErrorListener {
```

Comment:
This error listener allows us to rely less on non-default (Java built-in) Exception classes in a number of places, while still accurately reporting the problem. See https://github.com/checkstyle/checkstyle/pull/10280/files#r676140541